### DISCUSSION TITLE
basset & unpkg.com

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#1220] [Q&A (Help)] [answered]
For a few days now, but not all the time during the day, my Laravel + Backpack v6 apps are very slow and I see that the browser is waiting for responses from unpkg.com
I see that on ./vendor all basset commands use unpkg.com

Why ?
Does this happen to others ?
Solutions ?

Thanks for help

[a comment by an automated account is not shown]

**Comment by blondie63** [comment]
I've checked all points but the issue is still here !!
Why app call many times unpkg.com where basset is right configured ??

**Comment by karandatwani92** [comment, marked as answer]
Hey @blondie63 

Can you add `BASSET_DEV_MODE=false` to your env and run `php artisan basset:cache`?
This will force the app to cache the assets locally and server from the local cache.

**Reply by blondie63** [reply]
Now it's ok but it's strange that problem come only in last days..
Thanks

**Comment by yurik94** [comment]
Still slow, can we have an option to switch between another CDN service or to self host those files?
<img width="1064" alt="Screenshot 2024-12-23 alle 10 32 29" src="https://github.com/user-attachments/assets/4e46c0ef-f931-422f-b4bd-a07d9056ae6c" />

**Reply by yurik94** [reply]
@tabacitu resources from unpkg.com are again having problems.

We think that a better cdn should be used in the project.

![image](https://github.com/user-attachments/assets/9a3124fa-79a0-4c71-8c43-8ecd86aa1743)
![image](https://github.com/user-attachments/assets/cca11c27-f728-4fb0-8bca-8d6716b5554e)

**Reply by blondie63** [reply]
I've same problems but only locally on my mac, same code on dev server is fast !
<img width="253" alt="Screenshot 2025-03-15 alle 10 14 53" src="https://github.com/user-attachments/assets/ce298a31-5f4f-49e2-883e-9edea2dac2b4" />

what could it be?

**Reply by blondie63** [reply]
@tabacitu @yurik94 
I tried a thousand times and I discovered that the problem is definitely related to Basset and unpkg.com!
But if in my local .env I put app_env=production then everything works quickly!
The strange thing is that until a few days ago it didn't do this!!
What could it be?

**Reply by yurik94** [reply]
Hello Mauro @blondie63 

The problem is vey simple, the unpkg.com CDN is down/experiencing problems since a few days.

Status: [https://unpkg.com/status](https://unpkg.com/status)

As you can see from chrome devtools, some assets are loaded and some others are very slow to download/fail completely, probably because of CloudFlare trying to reach the origin server of unpkg.com which is down.

Since backpack is heavily relying on unpkg.com, which has a pretty bad reputation, I think that it should be changed with jsDeliver.


You can resolve temporarily by invoking: `sail artisan basset:cache`
With this command all of the bassets will be copied and served locally.

> Basset is disabled by default on local environment. If you want to change it, please set BASSET_DEV_MODE=false in your env file.

So you also need to put ~`BASSET_DEV_MODE=true`~ `BASSET_DEV_MODE=false` in your local .env

**Reply by yurik94** [reply]
Sorry, errata corrige:

`BASSET_DEV_MODE=false`

> enable/disable dev mode using BASSET_DEV_MODE=false - this will force Basset to internalize assets even on localhost

**Reply by tabacitu** [reply]
Yeah... this is really unfortunate... unpkg has really let us down here.mFor the moment, while they're having problems, that solution will work. 

In the next version, due to launch in a few weeks, we've already removed most unpkg links and we've provided a way to manually change the url to whatever you like, if the CDN proves unreliable.

But until then... BASSET_DEV_MODE=false will help.

**Comment by munjaldevelopment** [comment]
Getting `internal server error` on jquery unpkg.com. Facing issue on few projects due to this.

![image](https://github.com/user-attachments/assets/a6780066-88c9-4ebd-8d4c-70b0b5a07272)
